# HG changeset patch
# User Rik <rik@octave.org>
# Date 1433891804 25200
#      Tue Jun 09 16:16:44 2015 -0700
# Branch stable
# Node ID c4f436483e49180d1bb52b627fac3395d1d41d55
# Parent  209ee4a730f6cb6941d5d38e70ee9a350fface23
run.m: Return to original directory on Windows (bug #45231).

* run.m: Use make_absolute_filename, rather than canonicalize_file_name, to get
the original working directory in standard form.

diff --git a/scripts/miscellaneous/run.m b/scripts/miscellaneous/run.m
--- a/scripts/miscellaneous/run.m
+++ b/scripts/miscellaneous/run.m
@@ -56,17 +56,17 @@ function run (script)
 
   if (! exist (script, "file"))
     error ("run: file SCRIPT must exist and be a valid Octave scriptfile");
   endif
 
   if (! isempty (d))
     if (exist (d, "dir"))
       startdir = pwd ();
-      d = canonicalize_file_name (d);
+      d = make_absolute_filename (d);
       unwind_protect
         cd (d);
         evalin ("caller", sprintf ("source ('%s%s');", f, ext),
                 "rethrow (lasterror ())");
       unwind_protect_cleanup
         if (strcmp (d, pwd ()))
           cd (startdir);
         endif
